torch/_inductor/select_algorithm.py

```diff
@@ -1059,50 +1191,49 @@ def __init__(self, name: str, grid: Any, source: str, debug=False) -> None:
         self.grid = grid
         self.template = self._template_from_string(source)
         assert name not in self.all_templates, "duplicate template name"
-        self.all_templates[name] = self
+        TritonTemplate.all_templates[name] = self
         self.debug = debug
+        self._generated_module_cache: GeneratedModulesCache = GeneratedModulesCache()
+        clear_on_fresh_inductor_cache(self._generated_module_cache)
 
-    def generate(  # type: ignore[override]
+    # Those class fields are used for testing _generated_module_cache.
+    # When this flag is on, we ensure that the cached results and the generated result if cache
+    # was not used are the same.
+    test_cache = False
+    generated_module_cache_hit = 0
```

Comment:
Might be useful to have cache misses as well. (10 hits - yay - is that good? Out of 10 attempts? Yes.  Out of 100k? No.)